perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -4948,4 +4948,61 @@ private boolean isMemberInGracePeriod(Map<String, String> membershipExpirationRu
 	public List<Facility> getFacilitiesWhereGroupIsAdmin(PerunSession perunSession, Group group) throws InternalErrorException {
 		return this.getGroupsManagerImpl().getFacilitiesWhereGroupIsAdmin(perunSession, group);
 	}
+
+	@Override
+	public void reactivateMember(PerunSession sess, Member member, Group group) throws InternalErrorException, MemberNotExistsException {
+		if (!isGroupMember(sess, group, member)) throw new MemberNotExistsException("Member does not belong to this group");
+
+		validateMemberInGroup(sess, member, group);
+
+		Attribute membershipExpirationRulesAttribute = getMembershipExpirationRulesAttribute(sess, group);
```

Comment:
I just wonder, if setting group membership expiration to null and calling extendMembershipInGroup() wouldn't do the same job, so we could prevent code duplicity?

Reply:
We should definitely prevent code duplicity if possible.

Reply:
Yes, it would, thank you, it should be fixed now.